server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -59,6 +60,9 @@ public class RemoteFsTranslog extends Translog {
     // clean up translog folder uploaded by previous primaries once
     private final SetOnce<Boolean> olderPrimaryCleaned = new SetOnce<>();
 
+    // Semaphore used to allow only single remote generation to happen at a time
+    private final Semaphore remoteGenerationDeletionPermits = new Semaphore(2);
```

Comment:
As we create one instance of `RemoteFsTranslog` per shard, this semaphore is at shard level. Is this understanding correct?

Reply:
Yes, this is per shard. We have a higher pool to support parallel deletes across different shards.